src/core/quantum_gate.jl

```diff
@@ -154,6 +154,13 @@ function apply_operator!(
     end
 end
 
+function apply_operator!(
+    state::Ket,
+    operator::SparseOperator,
+    connected_qubit::Vector{<:Integer})
+    @warn "apply_operator! is not implemented for SparseOperator. Try using DenseOperator instead."
```

Comment:
@pleroux0  are we ok with pushing this to main?

Reply:
This code is dangerous. A warning is easy to miss, and it will lead to incorrect unexpected results. Rather remove it and let it crash or throw an error with a detailed message.